### ISSUE TITLE
CI should check appropriate copyright headers are in place

### ISSUE THREAD

**Issue by davecheney** [projectcontour/contour#1917]
To ensure we don’t create source files without the appropriate copyright messages.

**Comment by SDBrett** [comment]
GCP has a go test which might be helpful here

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/license_test.go

**Comment by SDBrett** [comment]
I have been unable to find a Github feature that can be leveraged for this task. So I've worked with the above sample from GCP.

I've made some updates to the above licence_test file, to allow filtering out of any go files within the generated directories as they are always flagging as missing copyright.

The is picking up a couple of error files, but does trigger on files where the top line is not the copyright statement. Typically this is files where the first line is `// +build`. 

This could be build into a standalone binary and run from a docker container or as a serverless function.

If this is a path you want to look at further, I would need assistance from those more experienced with Go.
